runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCacheCursorFactory.java

```diff
@@ -1001,7 +1090,14 @@ public KafkaFilterCondition asCondition(
         {
             final List<KafkaFilterCondition> asConditions = new ArrayList<>();
             filters.forEach(f -> asConditions.add(asCondition(f)));
-            condition = asConditions.size() == 1 ? asConditions.get(0) : new KafkaFilterCondition.Or(asConditions);
+            if (KafkaEvaluation.EAGER == evaluation)
+            {
+                condition = asConditions.size() == 1 ? asConditions.get(0) : new KafkaFilterCondition.EagerOr(asConditions);
+            }
+            else
+            {
+                condition = asConditions.size() == 1 ? asConditions.get(0) : new KafkaFilterCondition.Or(asConditions);
+            }
```

Comment:
Suggest changing this to `switch` on `evaluation`.

Note: for the optimization of single condition to still work, we would need to control the return value of `.test(...)` for the nested conditions, so they return the right bitmask when successful.

This approach may be helpful in general when implementing `EagerOr` and `LazyOr` as you can just bitwise-or the return values of `.test(...)` together to get the combined bitmask.